amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/AmqpBrokerDecoder.java

```diff
@@ -47,13 +47,18 @@ public ServerMethodProcessor<? extends ServerChannelMethodProcessor> getMethodPr
     @Override
     public void decodeBuffer(QpidByteBuffer buf) throws AMQFrameDecodingException, AMQProtocolVersionException,
             IOException {
-        if (netInputBuffer.remaining() < buf.remaining()) {
+        int messageSize = buf.remaining();
+        if (netInputBuffer.remaining() < messageSize) {
             QpidByteBuffer oldBuffer = netInputBuffer;
-            netInputBuffer = QpidByteBuffer.allocateDirect(bufferSize);
+            boolean oldBufferFlag = false;
             if (oldBuffer.position() != 0) {
                 oldBuffer.limit(oldBuffer.position());
                 oldBuffer.slice();
                 oldBuffer.flip();
+                oldBufferFlag = true;
+            }
+            netInputBuffer = QpidByteBuffer.allocateDirect(bufferSize + oldBuffer.remaining() + messageSize);
+            if (oldBufferFlag) {
```

Comment:
need to release the old buffer?

Reply:
If the position of the oldBuffer is 0 and the remaining of the oldBuffer is not 0, we also need to put it in the netInputBuffer, right?